### PR TITLE
External Secrets Operator and CLI

### LINKED ISSUE #2239: External-Secrets: Add supported providers to external-secrets docs.
This work should be done for TAP 1.5

The supported providers in external-secrets should be present, adding a better understanding of the options we support for creating and listing secrets.

Upstream docs for reference: https://external-secrets.io/v0.7.2/introduction/stability-support/

### PR DESCRIPTION
**Pull request by rashedkvm**
# Which other branches do you want a technical writer to cherry-pick this PR to (if any)?
`main` branch only (TAP 1.5.x)
It's best to PR to the most recent relevant branch and leave all the cherry-picking to the
writer, except where earlier branches require factual changes to the content.
For more information about the branches, see https://github.com/pivotal/docs-tap#branches

Fixes #2239

### CODE REVIEW

**Comment by snneji** [comment]
Hi @rashedkvm, is this ready to go yet? We only have a couple of weeks until 1.5 and I'll need time to edit your changes.

**Comment by rashedkvm** [comment]
> Hi @rashedkvm, is this ready to go yet? We only have a couple of weeks until 1.5 and I'll need time to edit your changes to meet VMware's style standards.

Hi @snneji, this is not ready yet. I expect to have this completed by this week.

**Comment by snneji** [comment]
Hi @rashedkvm, just to warn you, I might not have time to do your PR before release day as there's only just over a week left and we've had a lot of work come in recently. I will make sure it's merged, edited, and published within seven business days of the release (per our [Working Agreement](https://confluence.eng.vmware.com/pages/viewpage.action?pageId=1452304368)).

Just a tip, if you want to guarantee your changes make it into the docs on release day I recommend you submit your PR before bug bash. 🙂

File: external-secrets/about-external-secrets-operator.hbs.md [hunk 1]
```diff
@@ -1,182 +1,27 @@
-# External Secrets Operator (alpha)
+# External Secrets Operator (beta)
 
->**Caution** External Secrets Operator is in alpha and is intended for
->evaluation and test purposes only. Do not use in a production environment.
+>**Caution** External Secrets plugin available in Tanzu CLI list of plug-ins is in beta
```
Comment (rszumlakowski) [inline external-secrets/about-external-secrets-operator.hbs.md:3]:
```suggestion
>**Caution** The External Secrets plugin available in the Tanzu CLI list of plug-ins is in beta 
```

File: external-secrets/about-external-secrets-operator.hbs.md [hunk 2]
```diff
@@ -1,182 +1,27 @@
...
+>**Caution** External Secrets plugin available in Tanzu CLI list of plug-ins is in beta 
+and is intended for evaluation and test purposes only. Do not use it in a production environment.
 
 The [External Secrets Operator](https://external-secrets.io) is a Kubernetes
 operator that integrates with external secret management systems, for example,
 Google Secrets Manager and Hashicorp Vault. It reads information from external
-APIs and automatically injects the values into a Kubernetes secret. Starting
-with v1.4.0, Tanzu Application Platform repackages the External Secrets Operator
-v0.6.1 into a Carvel bundle.
+APIs and automatically inject the values into a Kubernetes secret. Tanzu Application Platform makes
```
Comment (rszumlakowski) [inline external-secrets/about-external-secrets-operator.hbs.md:9]:
```suggestion
APIs and automatically injects the values into Kubernetes secrets. Tanzu Application Platform makes
```

File: external-secrets/vault-example.hbs.md [hunk 3]
```diff
@@ -0,0 +1,139 @@
+# External Secrets Operator example Integration with HashiCorp Vault 
+
+>**IMPORTANT** This example integration is deliberately constructed to showcase the features available
+and must not be considered in a production environment. 
+
+This topic describes how External Secrets Operator integrates with HashiCorp Vault, an external 
+Secret Management System. The operator synchronizes secret data from external APIs to 
+[Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret).
```
Comment (rszumlakowski) [inline external-secrets/vault-example.hbs.md:8]:
```suggestion
[Kubernetes Secret](https://kubernetes.io/docs/concepts/configuration/secret) resources.
```

File: external-secrets/vault-example.hbs.md [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+   ```sh
+   tanzu eso secret list
+
+   NAMESPACE  NAME                  SECRET NAME      STORE               REFRESH INTERVAL  STATUS             LAST UPDATED  LAST REFRESH  
+   default    vault-secret-example  registry-secret  vault-secret-store  15m               SecretSynced  21s           292y   
+   ```
+
+6. A Kubernetes `secret` will be created after the `ExternalSecret` resource is successfully 
+reconciled. Check for a secret named `registry-secret` created by the referenced `ExternSecret`.
```
Comment (rszumlakowski) [inline external-secrets/vault-example.hbs.md:132]:
```suggestion
reconciled. Check for a secret named `registry-secret` created by the referenced `ExternalSecret`.
```

**Review by rszumlakowski — CHANGES_REQUESTED** [review]

File: external-secrets/about-external-secrets-operator.hbs.md (hunk 2 above)
Comment (rashedkvm) [reply external-secrets/about-external-secrets-operator.hbs.md:9]: `a secret` - is important in this context.

File: external-secrets/vault-example.hbs.md [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+see [External Secrets Operator](install-external-secrets-operator.hbs.md).
+
+- Install Tanzu CLI. Tanzu CLI includes the plug-in `external-secrets`. For Tanzu CLI installation, 
+see [Tanzu CLI](../install-tanzu-cli.hbs.md)
+
+- A running instance of HashiCorp Vault. In this instance, there will be a secret defined with a key `eso-demo/reg-cred`
+
+## <a id='eso-vault-setup'></a> Setup
+
+1. Create `secret` with the vault token
```
[the remark this answers is not in the document]
Comment (rashedkvm) [reply external-secrets/vault-example.hbs.md:24]: lower case `secret` is ok

File: external-secrets/vault-example.hbs.md [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+         auth:
+           tokenSecretRef:
+             name: "vault-token" # vault-token created in the previous step
+             key: "token"
+   EOF
+   ```
+
+3. Check the status of the `SecretStore` resource. Status should be `Valid`
+
+   Run the fp=llowing command to see the `SecretStore` status:
```
[the remark this answers is not in the document]
Comment (rashedkvm) [reply external-secrets/vault-example.hbs.md:70]: Wow - this one is out of the world :)

File: external-secrets/vault-example.hbs.md [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+   tanzu external-secrets secret list
+   ```
+
+   For examample:
+
+   ```sh
+   tanzu eso secret list
+
+   NAMESPACE  NAME                  SECRET NAME      STORE               REFRESH INTERVAL  STATUS             LAST UPDATED  LAST REFRESH  
+   default    vault-secret-example  registry-secret  vault-secret-store  15m               SecretSynced  21s           292y
```
[the remark this answers is not in the document]
Comment (rashedkvm) [reply external-secrets/vault-example.hbs.md:128]: It is by design from the upstream implementation. A `null` in the `LAST REFRESH` value is printed this odd way. We will need to handle this on the client side to suppress this odd value.

File: external-secrets/about-external-secrets-operator.hbs.md (hunk 2 above)
Comment (rszumlakowski) [reply external-secrets/about-external-secrets-operator.hbs.md:9]: I have updated my suggestion.

File: external-secrets/vault-example.hbs.md [hunk 8]
```diff
@@ -0,0 +1,139 @@
...
+   Run the fp=llowing command to see the `SecretStore` status:
+
+   ```sh
+   tanzu external-secrets store list
+   ```
+
+   For examample:
+
+   ```sh
+   tanzu eso store list
```
[the remark this answers is not in the document]
Comment (rszumlakowski) [reply external-secrets/vault-example.hbs.md]: I would be comfortable if there was a section that said something like "The `external-secrets` plugin has two aliases: `es` and `eso` but in this document we will always refer to it as `external-secrets`."

**Comment by rszumlakowski** [comment]
Looks good to me.

**Review by rszumlakowski — APPROVED** [review]

**Comment by kcoriordan** [comment]
@snneji will merge this

**Comment by vishnarayan** [comment]
Looks good to me. @snneji Good to merge from my end too

**Pull request merged**